app/cli/internal/action/attestation_init.go

```diff
@@ -103,21 +103,37 @@ func (action *AttestationInit) Run(ctx context.Context, opts *AttestationInitRun
 
 	// 0 - find or create the contract if we are creating the workflow (if any)
 	contractRef := opts.NewWorkflowContractRef
-	_, err := NewWorkflowDescribe(action.ActionsOpts).Run(ctx, opts.WorkflowName, opts.ProjectName)
-	if err != nil && status.Code(err) == codes.NotFound {
-		// Not found, let's see if we need to create the contract
-		if contractRef != "" {
-			// Try to find it by name
-			_, err := NewWorkflowContractDescribe(action.ActionsOpts).Run(contractRef, 0)
-			// An invalid argument might be raised if we use a file or URL in the "name" field, which must be DNS-1123
-			// TODO: validate locally before doing the query
-			if err != nil && (status.Code(err) == codes.NotFound || status.Code(err) == codes.InvalidArgument) {
-				createResp, err := NewWorkflowContractCreate(action.ActionsOpts).Run(fmt.Sprintf("%s-%s", opts.ProjectName, opts.WorkflowName), nil, contractRef)
+	// Not found, let's see if we need to create the contract
+	if contractRef != "" {
+		// Try to find it by name
+		_, err := NewWorkflowContractDescribe(action.ActionsOpts).Run(contractRef, 0)
+		// An invalid argument might be raised if we use a file or URL in the "name" field, which must be DNS-1123
+		if err != nil && (status.Code(err) == codes.NotFound || status.Code(err) == codes.InvalidArgument) {
```

Comment:
I find this hack not ideal, should we just check if it's a file or HTTP?

Reply:
fixed.